src/server/shared/SharedDefines.h

```diff
@@ -3770,6 +3770,15 @@ enum LineOfSightChecks
     LINEOFSIGHT_ALL_CHECKS      = (LINEOFSIGHT_CHECK_VMAP | LINEOFSIGHT_CHECK_GOBJECT)
 };
 
+enum ServerProcessTypes
+{
+    SERVER_PROCESS_AUTHSERVER = 0,
+    SERVER_PROCESS_WORLDSERVER = 1,
+
+    NUM_SERVER_PROCESS_TYPES
+};
+extern TC_SERVERS_API ServerProcessTypes THIS_SERVER_PROCESS;
```

Comment:
No, don't do it like that. Make it internal to shared, and export a function that you will call in authserver and worldserver to initialize to proper value

Reply:
what's the benefit?

Reply:
Perhaps I wasn't clear, you have to do it this way or it will not compile with dynamic linking. (see appveyor)